--- node_modules/@cubejs-backend/api-gateway/dist/src/gateway.js	2025-05-09 16:13:04
+++ node_modules/@cubejs-backend/api-gateway/dist/src/gateway.js	2025-05-09 16:13:04
@@ -1528,6 +1528,10 @@
     }
     resToResultFn(res) {
         return async (message, { status } = {}) => {
+            if (res?.req?.method == 'GET' && res.req?.route?.path?.startsWith('/cubejs-api/v1/load') && message?.error == undefined) {
+                res.set('Cache-Control', 'public, max-age=300, stale-while-revalidate=120, stale-if-error=1800')
+            }
+
             if (status) {
                 res.status(status);
             }
