--- node_modules/@cubejs-backend/query-orchestrator/dist/src/orchestrator/RedisQueueDriver.js	2024-03-26 14:09:49
+++ node_modules/@cubejs-backend/query-orchestrator/dist/src/orchestrator/RedisQueueDriver.js	2024-03-26 14:09:49
@@ -196,6 +196,11 @@
                 .set(lockKey, processingId, 'NX')
                 .zadd([this.heartBeatRedisKey(), 'NX', new Date().getTime(), this.redisHash(queryKey)])
                 .execAsync();
+
+            if (!result) {
+              return null;
+            }
+
             if (result) {
                 result[4] = JSON.parse(result[4]);
                 if (this.getQueueEventsBus) {
